### DISCUSSION TITLE
[v15] Make `IDetailedTelemetryProvider` async

### DISCUSSION THREAD

**Discussion by mattbrailsford** [umbraco/Umbraco-CMS#17369] [Features and ideas]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

v15

### Bug summary

In v15 Umbraco Commerce has gone fully async however in order to setup a `IDetailedTelemetryProvider` we have to force our code to be sync. `IDetailedTelemetryProvider` should have methods for collecting data async.

### Steps to reproduce

Self explanitory

### Expected result / actual result

I'd like to be able to use async services in a `IDetailedTelemetryProvider` implementation

**Comment by bergmania** [comment]
I like the idea. Would be a nice addition and should be possible to make in a backward compatible way
